beard/metrics/string_methods.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Beard.
+# Copyright (C) 2015 CERN.
+#
+# Beard is a free software; you can redistribute it and/or modify it
+# under the terms of the Revised BSD License; see LICENSE file for
+# more details.
+
+"""String metrics.
+
+.. codeauthor:: Petros Ioannidis <petros.ioannidis91@gmail.com>
+
+"""
+
+import re
+
+
+def find_all(s, pattern):
+    """Find all occurences of the given pattern.
+
+    Parameters
+    ----------
+    :param s: string
+        String to be searched
+
+    :param letter: string
+        Substring we are searching for
+
+    Returns
+    -------
+    :returns: generator
+        A generator that holds the indexes of the patterns
+    """
+    for match in re.finditer(pattern, s):
+            yield match.start()
+
+def jaro_matching(s1, s2):
+    """Return the number of matching letters based on the jaro algorithm.
```

Comment:
Strictly speaking, there is no "jaro algorithm", but only a "jaro similarity metric" . So I would rephrase this as "Return the number of matching letters in the two strings"